fetch_api/src/fetch_api/graspcli.py

```diff
@@ -0,0 +1,161 @@
+               obj.object.primitives[0].dimensions[0] > 0.14 or \
+               obj.object.primitives[0].dimensions[0] < 0.02 or \
+               obj.object.primitives[0].dimensions[0] > 0.14:
+                continue
+            # has to be on table
+            print("z")
+            print(obj.object.primitive_poses[0].position.z)
+            if obj.object.primitive_poses[0].position.z < 0.2:
+                continue
+            return obj.object, obj.grasps
+        # nothing detected
+        return None, None
+
+    # def getCokeReference(self, obj):
+    #     pose = obj.object.primitive_poses[0]
+    #     pose_stamped = PoseStamped()
+    #     pose_stamped.pose = pose
+    #     transform = tf_buffer.lookup_transform("map",
+    #                                    "head_camera_link", #source frame
+    #                                    rospy.Time(0), #get the tf at first available time
+    #                                    rospy.Duration(1.0)) #wait for 1 second
+    #     pose_transformed = do_transform_pose(pose_stamped, transform)
+    #     print pose_transformed
+    def getSupportSurface(self, name):
+        for surface in self.support_surfaces:
+            if surface.name == name:
+                return surface
+        return None
+
+    def getPlaceLocation(self):
+        pass
+
+    def pick(self, block, grasps):
+        success, pick_result = self.pickplace.pick_with_retry(block.name,
+                                                              grasps,
+                                                              support_name=block.support_surface,
+                                                              scene=self.scene)
+        self.pick_result = pick_result
+        return success
+
+    def place(self, block, pose_stamped):
+        places = list()
+        l = PlaceLocation()
+        l.place_pose.pose = pose_stamped.pose
+        l.place_pose.header.frame_id = pose_stamped.header.frame_id
+
+        # copy the posture, approach and retreat from the grasp used
+        l.post_place_posture = self.pick_result.grasp.pre_grasp_posture
+        l.pre_place_approach = self.pick_result.grasp.pre_grasp_approach
+        l.post_place_retreat = self.pick_result.grasp.post_grasp_retreat
+        places.append(copy.deepcopy(l))
+        # create another several places, rotate each by 360/m degrees in yaw direction
+        m = 16 # number of possible place poses
+        pi = 3.141592653589
+        for i in range(0, m-1):
+            l.place_pose.pose = rotate_pose_msg_by_euler_angles(l.place_pose.pose, 0, 0, 2 * pi / m)
+            places.append(copy.deepcopy(l))
+
+        success, place_result = self.pickplace.place_with_retry(block.name,
+                                                                places,
+                                                                scene=self.scene)
+        return success
+
+    def tuck(self):
```

Comment:
This would be good to have on `Arm`

Reply:
agreed, I was surprised we didnt already have it